inst/scripts/2.4.0/make-hermann-spermatogenesis-data.Rmd

```diff
@@ -0,0 +1,205 @@
+---
+title: Processing the Hermann spermatogenesis dataset for RNA velocity analysis
+author: Charlotte Soneson
+date: "June 5, 2020"
+output:
+  BiocStyle::html_document:
+    titlecaps: false
+    toc_float: true
+bibliography: ../ref.bib
+---
+
+```{r style, echo=FALSE, results='hide', message=FALSE}
+library(BiocStyle)
+knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE, eval=FALSE)
+```
+
+# Loading required R packages
+
+```{r}
+suppressPackageStartupMessages({
+    library(Biostrings)
+    library(BSgenome)
+    library(eisaR)
+    library(GenomicFeatures)
+    library(SummarizedExperiment)
+    library(tximeta)
+    library(rjson)
+    library(SingleCellExperiment)
+    library(dplyr)
+})
+```
+
+# Setting the working directory
+
+```{r}
+workdir <- tempdir()
+Sys.setenv(workdir = workdir)
+```
+
+# Downloading the reference sequences
+
+In order to perform the quantification of spliced and unspliced abundances with alevin [@srivastava2019alevin], we need to first prepare a suitable set of reference features (transcript isoform and intron sequences) to generate an index from. 
+We start by downloading the mouse reference genome and annotation gtf file from GENCODE.
+
+```{bash}
+wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/gencode.vM24.annotation.gtf.gz -P $workdir
+wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/GRCm38.primary_assembly.genome.fa.gz -P $workdir
+gunzip $workdir/GRCm38.primary_assembly.genome.fa.gz
+```
+
+# Extracting the transcript and intron sequences
+
+Next, we extract a GRanges object containing the genomic coordinates of each annotated transcript and intron. 
+
+```{r}
+gtf <- file.path(workdir, "gencode.vM24.annotation.gtf.gz")
+grl <- eisaR::getFeatureRanges(
+    gtf = gtf,
+    featureType = c("spliced", "intron"), 
+    intronType = "separate", 
+    flankLength = 90L, 
+    joinOverlappingIntrons = FALSE, 
+    verbose = TRUE
+)
+```
+
+After defining the genomic positions of all features of interest, we can extract the sequences of these, and write to a fasta file for later indexing with _Salmon_ [@patro2017salmon].
+In addition, we generate reference files that will be helpful for reading the data with _tximeta_ [@love2020tximeta] and splitting the joint count matrix into a spliced and an unspliced one.
+
+```{r}
+genome <- Biostrings::readDNAStringSet(
+    file.path(workdir, "GRCm38.primary_assembly.genome.fa")
+)
+names(genome) <- sapply(strsplit(names(genome), " "), .subset, 1)
+seqs <- GenomicFeatures::extractTranscriptSeqs(
+    x = genome, 
+    transcripts = grl
+)
+Biostrings::writeXStringSet(
```

Comment:
If you're going to do this anyway, why not generate FASTA files from an appropriate **BSgenome** package?

Reply:
I do it like this since I prefer to get all annotation files from the same source (in this case Gencode) to be sure that they are fully compatible. I anyway have to write the genome to a FASTA file for Salmon.